integrations/helm/git/git.go

```diff
@@ -98,9 +100,10 @@ func (ch *Checkout) Clone(ctx context.Context, cloneSubdir string) error {
 	}
 	ch.Dir = repoDir
 
-	repo, err := gogit.PlainClone(repoDir, false, &gogit.CloneOptions{
-		URL:  ch.Config.URL,
-		Auth: ch.auth,
+	repo, err := gogit.CloneContext(ctx, memory.NewStorage(), osfs.New(repoDir), &gogit.CloneOptions{
```

Comment:
Another possibility would be to use gogit.PlainCloneContext and have all info on the filesystem, freeing the memory for other uses. I don't have strong feelings about it though.

As we are cloning the correct branch, do we need the wt.Checkout (original lines 114-120)?